iconfetcher_test.go

```diff
@@ -0,0 +1,120 @@
+package jujusvg
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/charm.v5-unstable"
+)
+
+type IconFetcherSuite struct{}
+
+var _ = gc.Suite(&IconFetcherSuite{})
+
+var (
+	ts        *httptest.Server
+	tsIconUrl func(*charm.Reference) string
+)
+
+func (s *IconFetcherSuite) SetUpTest(c *gc.C) {
+	ts = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "<svg></svg>")
+	}))
+
+	tsIconUrl = func(ref *charm.Reference) string {
+		return ts.URL + "/" + ref.Path() + ".svg"
+	}
+}
+
+func (s *IconFetcherSuite) TearDownTest(c *gc.C) {
+	ts.Close()
+}
+
+func (s *IconFetcherSuite) TestFetchIcons(c *gc.C) {
+	b, err := charm.ReadBundleData(strings.NewReader(bundle))
+	c.Assert(err, gc.IsNil)
+	err = b.Verify(nil)
+	c.Assert(err, gc.IsNil)
+	fetcher := HttpFetcher{
+		FetchConcurrently: false,
+		IconURL:           tsIconUrl,
+	}
+	iconMap, err := fetcher.FetchIcons(b)
+	c.Assert(err, gc.IsNil)
+	c.Assert(iconMap, gc.DeepEquals, map[string]string{
+		"~charming-devs/precise/elasticsearch-2": "<svg></svg>\n",
```

Comment:
This test would be more convincing if there was different content for each different charms. Also, it would be good if the bundle contained several services using the same charm, so that the "only fetch once" logic is tested. (even better if we actually check that the HTTP handler is invoked exactly once for each URL)